server/db/PostManager.ts

```diff
@@ -152,10 +153,41 @@ export class PostManager {
 			date,
 			...(parent && { parent })
 		});
+
+		// send a notification to all posts above, that are not made by this user
+		this.notifyPostChain(post._id, username);
+		subscribeUser(username, post._id);
 		await userManager.increasePostCounter(username);
 		return post._id;
 	}
 
+	async notifyPostChain(startId: mongoose.Types.ObjectId, username: string) {
+		const matchStartId = new ObjectId(startId);
+		const posts = await this.model
+			.aggregate()
+			.match({
+				_id: matchStartId
+			})
+			.graphLookup({
+				from: 'posts',
+				startWith: '$parent',
+				connectFromField: 'parent',
+				connectToField: '_id',
+				as: 'chain'
+			})
+			.project({ chain: 1 })
+			.unwind('chain')
+			.sort({ 'chain.date': -1 })
```

Comment:
Is the sorting necessary? Can we avoid it?

Reply:
I think it is necessary for the grouping stage as we want to pick the first "deepest" post from the user which would also be the latest. It may work without it but according to mongoose "Documents returned in the as field are not guaranteed to be in any order." for graphLookup so we have to do this sorting.